packages/discord.js/src/managers/GuildScheduledEventManager.js

```diff
@@ -276,8 +277,26 @@ class GuildScheduledEventManager extends CachedManager {
 
     let { limit, withMember, before, after } = options;
 
-    const data = await this.client.api.guilds(this.guild.id, 'scheduled-events', guildScheduledEventId).users.get({
-      query: { limit, with_member: withMember, before, after },
+    const query = new URLSearchParams();
+
+    if (limit) {
+      query.set('limit', limit);
+    }
+
+    if (withMember) {
+      query.set('with_member', withMember);
+    }
+
+    if (before) {
+      query.set('before', before);
+    }
+
+    if (after) {
+      query.set('after', after);
+    }
+
+    const data = await this.client.rest.get(Routes.guildScheduledEventUsers(this.guild.id, guildScheduledEventId), {
+      query: Array.from(query).length > 0 ? query : undefined,
```

Comment:
Is this needed?

Reply:
we automatically append ? to the url if query is set even if it ends up being an empty string. I *think* that won't be a problem for discord but thought better not to send it

Reply:
We should probably handle that in rest to only append `?${queryString}` if its not empty imo

Reply:
👍 kinda what I wanted to hear